### ISSUE TITLE
[7.6.1]: Android: APP Restarted After Foreground Push opened and close chat

### ISSUE THREAD

**Issue by davigsouza81** [deepak786/intercom_flutter#279]
https://user-images.githubusercontent.com/31492800/211059393-3a19c9a0-f585-4ec8-ba91-5edd0e6e27ac.mp4

```
I/ViewRootImpl@da41c4a[IntercomConversationActivity]( 4726): ViewPostIme pointer 0
I/ViewRootImpl@da41c4a[IntercomConversationActivity]( 4726): ViewPostIme pointer 1
I/DecorView( 4726): [INFO] isPopOver=false, config=true
I/DecorView( 4726): updateCaptionType >> DecorView@fe3b6c3[], isFloating=false, isApplication=true, hasWindowControllerCallback=true, hasWindowDecorCaption=false
D/DecorView( 4726): setCaptionType = 0, this = DecorView@fe3b6c3[]
I/DecorView( 4726): getCurrentDensityDpi: from real metrics. densityDpi=480 msg=resources_loaded
D/NativeCustomFrequencyManager( 4726): [NativeCFMS] BpCustomFrequencyManager::BpCustomFrequencyManager()
I/DecorView( 4726): notifyKeepScreenOnChanged: keepScreenOn=false
I/ViewRootImpl@e38e382[MainActivity]( 4726): setView = com.android.internal.policy.DecorView@fe3b6c3 TM=true
I/DecorView( 4726): [INFO] isPopOver=false, config=true
I/DecorView( 4726): updateCaptionType >> DecorView@24bf42c[], isFloating=false, isApplication=true, hasWindowControllerCallback=true, hasWindowDecorCaption=false
D/DecorView( 4726): setCaptionType = 0, this = DecorView@24bf42c[]
I/DecorView( 4726): getCurrentDensityDpi: from real metrics. densityDpi=480 msg=resources_loaded
D/NativeCustomFrequencyManager( 4726): [NativeCFMS] BpCustomFrequencyManager::BpCustomFrequencyManager()
I/ViewRootImpl@b6f6bd7[IntercomRootActivity]( 4726): setView = com.android.internal.policy.DecorView@24bf42c TM=true
I/SurfaceView@821746( 4726): onWindowVisibilityChanged(0) true io.flutter.embedding.android.FlutterSurfaceView{821746 V.E...... ......I. 0,0-0,0} of ViewRootImpl@e38e382[MainActivity]
I/ViewRootImpl@e38e382[MainActivity]( 4726): Relayout returned: old=(0,0,1080,2400) new=(0,0,1080,2400) req=(1080,2400)0 dur=5 res=0x7 s={true -5476376671465220096} ch=true fn=-1
D/OpenGLRenderer( 4726): eglCreateWindowSurface
I/SurfaceView@821746( 4726): windowStopped(false) true io.flutter.embedding.android.FlutterSurfaceView{821746 V.E...... ......ID 0,0-1080,2256} of ViewRootImpl@e38e382[MainActivity]
I/ViewRootImpl@e38e382[MainActivity]( 4726): [DP] dp(1) 1 android.view.ViewRootImpl.reportNextDraw:11442 android.view.ViewRootImpl.performTraversals:4198 android.view.ViewRootImpl.doTraversal:2924
I/SurfaceView@821746( 4726): pST: sr = Rect(0, 0 - 1080, 2256) sw = 1080 sh = 2256
I/SurfaceView@821746( 4726): onSSPAndSRT: pl = 0 pt = 0 sx = 1.0 sy = 1.0
I/SurfaceView@821746( 4726): pST: mTmpTransaction.apply, mTmpTransaction = android.view.SurfaceControl$Transaction@74543a
I/SurfaceView@821746( 4726): updateSurface: mVisible = true mSurface.isValid() = true
I/SurfaceView@821746( 4726): updateSurface: mSurfaceCreated = false surfaceChanged = true visibleChanged = true
I/SurfaceView@821746( 4726): surfaceCreated 1 #8 io.flutter.embedding.android.FlutterSurfaceView{821746 V.E...... ......ID 0,0-1080,2256}
I/SurfaceView@821746( 4726): surfaceChanged (1080,2256) 1 #8 io.flutter.embedding.android.FlutterSurfaceView{821746 V.E...... ......ID 0,0-1080,2256}
I/ViewRootImpl@e38e382[MainActivity]( 4726): [DP] dp(2) 1 android.view.SurfaceView.updateSurface:1375 android.view.SurfaceView.lambda$new$1$SurfaceView:254 android.view.SurfaceView$$ExternalSyntheticLambda2.onPreDraw:2
I/ViewRootImpl@e38e382[MainActivity]( 4726): [DP] pdf(1) 1 android.view.SurfaceView.notifyDrawFinished:599 android.view.SurfaceView.performDrawFinished:586 android.view.SurfaceView.$r8$lambda$st27mCkd9jfJkTrN_P3qIGKX6NY:0
D/ViewRootImpl@e38e382[MainActivity]( 4726): pendingDrawFinished. Waiting on draw reported mDrawsNeededToReport=1
D/InputTransport( 4726): Input channel destroyed: 'ClientS', fd=118
D/InputTransport( 4726): Input channel destroyed: 'ClientS', fd=103
I/ViewRootImpl@b6f6bd7[IntercomRootActivity]( 4726): Relayout returned: old=(0,0,1080,2400) new=(0,0,1080,2400) req=(1080,2400)0 dur=17 res=0x7 s={true -5476376671463712768} ch=true fn=-1
D/OpenGLRenderer( 4726): eglCreateWindowSurface
I/ViewRootImpl@b6f6bd7[IntercomRootActivity]( 4726): [DP] dp(1) 1 android.view.ViewRootImpl.reportNextDraw:11442 android.view.ViewRootImpl.performTraversals:4198 android.view.ViewRootImpl.doTraversal:2924
D/ViewRootImpl@b6f6bd7[IntercomRootActivity]( 4726): Creating frameDrawingCallback nextDrawUseBlastSync=false reportNextDraw=true hasBlurUpdates=false
D/ViewRootImpl@b6f6bd7[IntercomRootActivity]( 4726): Creating frameCompleteCallback
D/ViewRootImpl@b6f6bd7[IntercomRootActivity]( 4726): Received frameDrawingCallback frameNum=1. Creating transactionCompleteCallback=false
I/BufferQueueProducer( 4726): [ViewRootImpl@b6f6bd7[IntercomRootActivity]#8(BLAST Consumer)8](id:127600000008,api:1,p:4726,c:4726) queueBuffer: queued for the first time.
D/OpenGLRenderer( 4726): GPIS:: SetUp Pid : 4726    Tid : 4774
D/ViewRootImpl@b6f6bd7[IntercomRootActivity]( 4726): Received frameCompleteCallback  lastAcquiredFrameNum=1 lastAttemptedDrawFrameNum=1
I/ViewRootImpl@b6f6bd7[IntercomRootActivity]( 4726): [DP] pdf(0) 1 android.view.ViewRootImpl.lambda$addFrameCompleteCallbackIfNeeded$3$ViewRootImpl:5000 android.view.ViewRootImpl$$ExternalSyntheticLambda16.run:6 android.os.Handler.handleCallback:938
I/ViewRootImpl@b6f6bd7[IntercomRootActivity]( 4726): [DP] rdf()
D/ViewRootImpl@b6f6bd7[IntercomRootActivity]( 4726): reportDrawFinished (fn: -1)
I/ViewRootImpl@da41c4a[IntercomConversationActivity]( 4726): MSG_WINDOW_FOCUS_CHANGED 0 1
D/InsetsSourceConsumer( 4726): ensureControlAlpha: for ITYPE_NAVIGATION_BAR on br.com.myapp.example/io.intercom.android.sdk.m5.IntercomRootActivity
D/InsetsSourceConsumer( 4726): ensureControlAlpha: for ITYPE_STATUS_BAR on br.com.myapp.example/io.intercom.android.sdk.m5.IntercomRootActivity
3
D/FRCPlugin( 4726): Sending fetchTimeout: 10
I/flutter ( 4726): AppModule started!
```

**Comment by deepak786** [comment]
@davigsouza81 What is the `launchMode` of the MainActivity in the Android manifest?
Can you try with `singleTop` and see if the issue is resolved or not.

```
android:launchMode="singleTop"
```

**Comment by JycTheDev** [comment]
The issue still persists

**Comment by deepak786** [comment]
@jbankz Can you please share the reproducible sample?

**Comment by JycTheDev** [comment]
@deepak786 the same as the video sent above by @davigsouza81 ......i tried the solution you proposed earlier but still didn't work

**Comment by deepak786** [comment]
@jbankz I can see the same behavior as per my testing on the Android platform.

- Displayed the current timestamp on the screen.
- Opened the messenger from button click. Closed the messenger. Timestamp is not changed. So this is working fine.
- Opened the messenger from the foreground push notification banner at the bottom. Closed the messenger. The timestamp on the screen is now updated. Which means the app is restarted.

**_And I also confirm that the app restarts when you click back from the particular chat screen which was opened from the foreground push notification click. On back press, it shows you the messages screen and this is the point where the app restarts._**

The issue started appearing from the Intercom Android SDK version 14.0.0

I will report it to the Intercom support team.
Meanwhile could you please check if the same thing happens on the iOS platform or not?

**Comment by JycTheDev** [comment]
@deepak786 Thank you so much for your quick response, 
So far, the iOS platform works fine as expected.

**Comment by barees63** [comment]
Just to add to this, I'm also seeing this on Android but not on iOS.

**Comment by deepak786** [comment]
Update from Intercom Support

```
Hi Deepak,

I’ve heard back from our team and can confirm that we’re adding this issue to our backlog to be fixed!

For greater context, our product team prioritizes bugs based on the severity of the issue and how widespread of an experience it is with customers. To provide insight on expectations, issues like this may not be actioned for some months. That timeline is based on the team's current existing log of issues and upcoming feature releases which is subject to change.

Given the long-term nature of the work to be done here, I’m going to close this conversation with you for now, but I will reopen the case and send an update when the team moves forward with a fix 👍

I appreciate you taking the time to surface this issue with our team and you can rest assured I’ll take ownership from here!
```

**Comment by barees63** [comment]
That's a very weak response from them, Is there any workaround for this?  Can we temporarily disable push for Android?  I'm not able to reopen the app after this happens, crashes on startup as presumably the deeplink keeps getting sent to the app.

**Comment by deepak786** [comment]
Maybe you can disable the in-app messages https://pub.dev/documentation/intercom_flutter/latest/intercom_flutter/Intercom/setInAppMessagesVisibility.html

**Comment by barees63** [comment]
ok,  I'll see what I can do and if I find a workaround I'll post it here.  Thanks for the package BTW, I've been using it for a long time and it's been great.

**Comment by utopicnarwhal** [comment]
@deepak786 seems like they've fixed the issue according to changelog of 14.0.5
Source: https://github.com/intercom/intercom-android/pull/596/files

**Comment by JycTheDev** [comment]
@utopicnarwhal thats for Android native yea ?

**Comment by utopicnarwhal** [comment]
> @utopicnarwhal thats for Android native yea ?

Yes, we should wait till someone bumps the version in the flutter package

**Comment by deepak786** [comment]
The reply from the Intercom support was after the release date of Android SDK version 14.0.5 as the release date they mentioned is Feb 10, 2023.
See https://github.com/v3rm0n/intercom_flutter/issues/279#issuecomment-1431685433

But the PR to release the version is created 5 days ago https://github.com/intercom/intercom-android/pull/596/files

And also there is no update from the Intercom support on the ticket.
So I'm not sure if the issue is fixed by the SDK v14.0.5.
But I have already created a PR to use the latest SDK versions https://github.com/v3rm0n/intercom_flutter/pull/289.

Once this is merged and released, give it a try.
Maybe you can already test this from the master branch.

**Comment by deepak786** [comment]
`intercom_flutter: 7.6.5` is released that uses Android SDK v14.0.5

**Comment by utopicnarwhal** [comment]
You were right. It still restarts the app 😔

**Comment by vinhdn** [comment]
To resolve this issue in Android, I disabled in-app message:
`Intercom.instance.setInAppMessagesVisibility(IntercomVisibility.gone);`
And I handle show new message when receive a new one on:
``` dart
FirebaseMessaging.onMessage.listen((RemoteMessage message) async {
  if(message.data != null && 'intercom_sdk' == message.data['receiver']) {
    if(Platform.isAndroid) {
      ScaffoldMessenger.of(context).clearSnackBars();
      final body = message.data['body'];
      final snack = new SnackBar(
        dismissDirection: DismissDirection.horizontal,
        content:  InkWell(child: Text(body), onTap: () {
            Intercom.instance.displayMessages();
            ScaffoldMessenger.of(context).clearSnackBars();
        }
      );
      ScaffoldMessenger.of(context).showSnackBar(snack);
    }
    return;
  }
}
```
When user tap to `SnackBar`, I will display messages `Intercom.instance.displayMessages();`

> You can custom content in Snackbar same as your style.

**Comment by deepak786** [comment]
This has been resolved in Android SDK version [14.2.0](https://github.com/intercom/intercom-android/blob/master/CHANGELOG.md#1420)
